### ISSUE TITLE
WaterHeater:HeatPump:PumpedCondenser can crash when Outdoor Air Node name is required but is blank

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5747]
## Issue overview

This user file crashes because the outdoor air node number is zero in InitWaterThermalTank.  Since the inlet configuration is OutdoorAirOnly, the outdoor air node is required and this should be caught during input.  Should probably check other configurations (for both PumpedCondenser and WrappedCondenser) to be sure the required nodes are enforced as needed.

Note that the release version does not always crash with this error, but it does halt [here](https://github.com/NREL/EnergyPlus/blob/ae55b4e023f6ba80f4fb7f70d6b72a02932c77da/src/EnergyPlus/WaterThermalTanks.cc#L5199)  in the debugger.

Note with the OA node name fixed, this file has other node connection errors related to the HPWH.  Would it be possible to catch these earlier to provide more informative errors about what's wrong?  I'm guessing the fan and coil are in the wrong order, or there's some mismatch in node names.
### Details

Some additional details for this issue (if relevant):
- Platform (Win64)
- Version of EnergyPlus (v8.5.0-4c8aa42097)
- Helpdesk ticket 11443
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Defect file added (EnergyPlusDevSupport\DefectFiles)
- [x] Ticket added to Pivotal for defect (development team task)
- [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by rraustad** [comment]
Refactoring somehow eliminated checks for node names. Added back in but am checking on correct implementation.

```
** Severe  ** WaterHeater:HeatPump:PumpedCondenser="HPWH":
**   ~~~   ** When Inlet Air Configuration="OUTDOORAIRONLY".
**   ~~~   ** Outdoor Air Node Name and Exhaust Air Node Name must be specified.

** Severe  ** WaterHeater:HeatPump:PumpedCondenser="HPWATERHEATER_DHW":
**   ~~~   ** When Inlet Air Configuration="OUTDOORAIRONLY".
**   ~~~   ** Outdoor Air Node Name and Exhaust Air Node Name must be specified.
```

**Comment by rraustad** [comment]
I don't see any other issues in the defect file err log related to fan or coil placement? I'll check if a node check is needed to confirm correct placement of components.

```
** Warning ** GetDXCoils: Coil:WaterHeating:AirToWaterHeatPump:Pumped="AWHP_COIL":
**   ~~~   ** ...The Heating COP Function of Temperature Curve Name should be normalized to 1.0 at the rating point.
**   ~~~   ** ...Curve output at an air flow fraction of 1 = 1.075
**   ~~~   ** ...The simulation continues using the user-specified curve.
** Warning ** GetDXCoils: Coil:WaterHeating:AirToWaterHeatPump:Pumped="AWHP_COIL_DHW":
**   ~~~   ** ...The Heating COP Function of Temperature Curve Name should be normalized to 1.0 at the rating point.
**   ~~~   ** ...Curve output at an air flow fraction of 1 = 1.075
**   ~~~   ** ...The simulation continues using the user-specified curve.
** Warning ** WaterHeater:HeatPump:PumpedCondenser="HPWH": minimum inlet air temperature for heat pump compressor operation must be greater than or equal to -5 C.
**   ~~~   ** ...Minimum inlet air temperature = -15.0
** Warning ** WaterHeater:HeatPump:PumpedCondenser="HPWATERHEATER_DHW": minimum inlet air temperature for heat pump compressor operation must be greater than or equal to -5 C.
**   ~~~   ** ...Minimum inlet air temperature = -15.0
** Severe  ** Coil:WaterHeating:AirToWaterHeatPump:Pumped "AWHP_COIL_DHW": Rated air volume flow rate per watt of rated total water heating capacity is out of range
**   ~~~   ** Min Rated Vol Flow Per Watt=[4.027E-005], Rated Vol Flow Per Watt=[2.142E-004], Max Rated Vol Flow Per Watt=[8.056E-005]. See Input-Output Reference Manual for valid range.
** Severe  ** Coil:WaterHeating:AirToWaterHeatPump:Pumped "AWHP_COIL": Rated air volume flow rate per watt of rated total water heating capacity is out of range
**   ~~~   ** Min Rated Vol Flow Per Watt=[4.027E-005], Rated Vol Flow Per Watt=[1.705E-004], Max Rated Vol Flow Per Watt=[8.056E-005]. See Input-Output Reference Manual for valid range.
** Warning ** Chiller:Electric:EIR = CHILLER_EIRWC:  Integrated Part Load Value (IPLV) calculated is not at the AHRI test condition.
**   ~~~   **  Check limits in Cooling Capacity Function of Temperature Curve, Curve Type = BIQUADRATIC, Curve Name = WATER COOLED CHILLER CCFOFT
**   ~~~   **  Check limits in EIR Function of Temperature Curve, Curve Type = BIQUADRATIC, Curve Name = WATER COOLED CHILLER EITOCOOLINGOUTPUTRATIOFOFT
```

**Comment by rraustad** [comment]
Not sure if a specific test for node connections is required since the node checks should catch the problem later. Of course, this should probably be a fatal error if the components are not connected correctly.

Changing defect file to this:

```
Fan:OnOff,
    Fan_AWHP_DHW,            !- Name
    HPDHW Inlet,             !- Air Inlet Node Name
!    Fan_AWHP_DHW_OUT,        !- Air Outlet Node Name
    HPDHW Outlet,
```

Yields:

```
** Severe  ** Node Connection Error, Node="FAN_AWHP_DHW_OUT", Inlet node did not find an appropriate matching "outlet" node.
**   ~~~   ** If this is an outdoor air inlet node, it must be listed in an OutdoorAir:Node or OutdoorAir:NodeList object.
**   ~~~   ** Reference Object=COIL:WATERHEATING:AIRTOWATERHEATPUMP:PUMPED, Name=AWHP_COIL_DHW
** Severe  ** Node Connection Error, Node="HPDHW OUTLET", The same node appears as a non-parent Outlet node more than once.
**   ~~~   ** Reference Object=COIL:WATERHEATING:AIRTOWATERHEATPUMP:PUMPED, Name=AWHP_COIL_DHW
**   ~~~   ** Reference Object=FAN:ONOFF, Name=FAN_AWHP_DHW
```

**Comment by rraustad** [comment]
To catch the above input node name problem with the fan, I added a warning for the fan outlet node as:

```
** Severe  ** WaterHeater:HeatPump:PumpedCondenser="HPWATERHEATER_DHW":
**   ~~~   ** Heat pump water heater fan outlet node name not does not match next connected component.
**   ~~~   ** Fan outlet node name = HPDHW OUTLET
**   ~~~   ** Expected fan outlet node name = FAN_AWHP_DHW_OUT
**  Fatal  ** Errors found in getting WaterHeater:HeatPump:PumpedCondenser input. Preceding condition causes termination.
```

This still leaves the fan inlet node and coil nodes to test. I guess those tests are also needed since the input error could be anywhere.

**Comment by rraustad** [comment]
See PR #5754

**Comment by mjwitte** [comment]
Closed via #5754
